### ISSUE TITLE
Not able to change fields in globals.

### ISSUE THREAD

**Issue by teryhill** [LibreHealthIO/lh-ehr#1162]
I cannot change some of the fields in globals. I checked it on the demo site and I get the same issue. For example I am changing calendar end time and it will not save the new entry.

![image](https://user-images.githubusercontent.com/4655366/40754924-ddcd57f6-6449-11e8-9887-c12f9667c71b.png)

I think it is an issue with the field validation.

**Comment by aethelwulffe** [comment]
AM PM issue?

**Comment by apooravc** [comment]
Working on it, in Calendar globals, the starting hour & ending hour fields + last 5 fields also are not saving changes.

**Comment by apooravc** [comment]
@teryhill From line `205` to `320`, is the code for checking field values and updating them in DB. I think the issue with calendar starting/ending hour is that the flow of control never goes inside the corresponding `else if check` block at line `300`. The code iterates through each entry in `$GLOBALS_METADATA`, so it should go once inside `300` block as it does in other blocks above it. If it doesn't goes, then `$boolean` value would never be true and hence no update in DB by block at line `315` which is the case for starting and ending hour. Can't get why it's not going as `fldid = "schedule_end" or $fldid = "schedule_start"` in `globals.inc.php`.

The issue with rest field values might be that there is no similar (above) code for such values like there is for starting and ending hour. Like, for `Appointments - Patient Summary - Number to Display`, `fldid = number_of_appts_to_show` according to `globals.inc.php` but there is no such `else if check` for it in `add_edit_event.php`.

**Comment by teryhill** [comment]
schedule_start and schedule_end are the names of those fields in the metadata so I am not sure why those would not be working.  If the others are missing the checks that could be the reason they are not being changed.

**Comment by apooravc** [comment]
@teryhill Whenever I want to check where flow of control is going while program is running, I add a `echo "<script>console.log('{here}')</script>"`; to the blocks which I want to check. When I checked in starting and ending hour case, I found that there was no log in console for them, but there was a log for the blocks above them. I haven't checked other sections, but I think a lot are missing then?
`
